test/powershell/Language/Scripting/ActionPreference.Test.ps1

```diff
@@ -0,0 +1,129 @@
+﻿
+Describe "Tests for (error, warning, etc) action preference" -Tags "CI" {
+
+        Context 'Setting ErrorActionPreference to stop prevents user from getting the error exception' {
+            $err = $null
+            try
+            {
+                dir nosuchfile.nosuchextension -ea stop -ev err                
+            }
+            catch {}
+             
+            It '$err.Count' { $err.Count | Should Be 1 }
+            It '$err[0] should not be $null' { $err[0] | Should Not Be $null }
+            It '$err[0].GetType().Name' { $err[0].GetType().Name | Should Be ActionPreferenceStopException }
+            It '$err[0].ErrorRecord' { $err[0].ErrorRecord | Should not BeNullOrEmpty }
+            It '$err[0].ErrorRecord.Exception.GetType().Name' { $err[0].ErrorRecord.Exception.GetType().Name | Should Be ItemNotFoundException }
+        }
+        
+        It 'ActionPreference Ignore Works' {            
+            $errorCount = $error.Count
+            Get-Process -Name asdfasdfsadfsadf -ErrorAction Ignore
+            $errorCount2 = $error.Count
+
+            $error.Count | Should Be $errorCount
+        }
+        
+        Context 'action preference of Ignore cannot be set as a preference variable' {            
+            $failed = $true
+            try {
+                $GLOBAL:errorActionPreference = "Ignore"
```

Comment:
This won't be able to set the global state, error will occur and global state won't change.